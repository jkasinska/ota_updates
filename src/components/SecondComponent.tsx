import React from 'react';
import {SafeAreaView, SectionList, StyleSheet, Text, View} from 'react-native';

const DATA = [
  {
    title: "Main dishes",
    data: ["Pizza", "Burger", "Risotto"]
  },
  {
    title: "Sides",
    data: ["French Fries", "Onion Rings", "Fried Shrimps"]
  },
  {
    title: "Drinks",
    data: ["Water", "Coke", "Beer"]
  },
  {
    title: "Desserts",
    data: ["Cheese Cake", "Ice Cream"]
  }
];

const Item = (item: any) => (
  <View style={styles.item}>
    <Text style={styles.title}>{item.title}</Text>
  </View>
);

export default () => {
  return (
    <SafeAreaView style={styles.container}>
      <Text style={styles.componentTitle}>SecondComponent</Text>
      <SectionList
        sections={DATA}
        keyExtractor={(item, index) => item + index}
        renderItem={({ item }) => {
          console.log(item)
          return (
            <Item title={item} />
          );
        }}
        renderSectionHeader={({ section: { title } }) => (
          <Text style={styles.header}>{title}</Text>
        )}
      />
    </SafeAreaView>
  );
};

const styles = StyleSheet.create({
  container: {
    flex: 1,
    marginTop: 20,
    marginHorizontal: 16
  },
  item: {
    backgroundColor: "#f9c2ff",
    padding: 20,
    marginVertical: 8
  },
  header: {
    fontSize: 32,
  },
  title: {
    fontSize: 24
  },
  componentTitle: {
    fontSize: 15,
    textAlign: "center",
    marginTop: 10
  }
});
