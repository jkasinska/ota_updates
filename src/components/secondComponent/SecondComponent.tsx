import React, { useState } from 'react';
import {
  Button,
  SafeAreaView,
  SectionList,
  StyleSheet,
  Text,
  TextInput,
  View,
} from 'react-native';

const DATA = [
  {
    title: 'Main dishes',
    data: ['Pizza', 'Burger', 'Risotto'],
  },
  {
    title: 'Sides',
    data: ['French Fries', 'Onion Rings', 'Fried Shrimps'],
  },
  {
    title: 'Drinks',
    data: ['Water', 'Coke', 'Beer'],
  },
  {
    title: 'Desserts',
    data: ['Cheese Cake', 'Ice Cream'],
  },
];

const Item = (item: any) => (
  <View style={styles.item}>
    <Text style={styles.title}>{item.title}</Text>
  </View>
);

export default () => {
  const [value, changeValue] = useState('');
  const [isVisible, changeVisibility] = useState(false);

  return (
    <SafeAreaView style={styles.container}>
      <Text style={styles.componentTitle}>SecondComponent</Text>
      <View>
        <TextInput
          testID="input"
          value={value}
          onChangeText={changeValue}
          style={styles.textInput}
        />
        <Button
          color="#841584"
          title="Change label visibility"
          onPress={() => changeVisibility(!isVisible)}
        />
        {isVisible && <Text testID="value">{value}</Text>}
      </View>
      <SectionList
        sections={DATA}
        keyExtractor={(item, index) => item + index}
        renderItem={({ item }) => <Item title={item} />}
        renderSectionHeader={({ section: { title } }) => (
          <Text style={styles.header}>{title}</Text>
        )}
      />
    </SafeAreaView>
  );
};

const styles = StyleSheet.create({
  container: {
    flex: 1,
    marginTop: 20,
    marginHorizontal: 16,
  },
  item: {
    backgroundColor: '#F9C2FF',
    padding: 20,
    marginVertical: 8,
  },
  header: {
    fontSize: 32,
  },
  title: {
    fontSize: 24,
  },
  componentTitle: {
    fontSize: 15,
    textAlign: 'center',
    marginTop: 10,
  },
  textInput: {
    height: 40,
    borderColor: 'gray',
    borderWidth: 1,
  },
});
