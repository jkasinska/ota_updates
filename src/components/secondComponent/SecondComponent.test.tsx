import React from 'react';
import { fireEvent, render } from '@testing-library/react-native';
import SecondComponent from './SecondComponent';

describe('SecondComponent should', () => {
  test('match the snapshot', () => {
    const component = render(<SecondComponent />);
    expect(component).toMatchSnapshot();
  });

  test('change label visibility on button click', () => {
    const component = render(<SecondComponent />);
    const { getByTestId, getByText, queryByTestId, toJSON } = component;
    const newInputValue = 'test test';

    const input = getByTestId('input');
    fireEvent.changeText(input, newInputValue);

    const button = getByText('Change label visibility');
    fireEvent.press(button);

    expect(queryByTestId('value')).toBeTruthy();
    expect(getByTestId('value').props.children).toBe(newInputValue);
    expect(toJSON).toMatchSnapshot();
  });
});
