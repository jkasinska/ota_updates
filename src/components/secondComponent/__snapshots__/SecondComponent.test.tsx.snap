// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SecondComponent should change label visibility on button click 1`] = `[Function]`;

exports[`SecondComponent should match the snapshot 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Object {
      "flex": 1,
      "marginHorizontal": 16,
      "marginTop": 20,
    }
  }
>
  <Text
    style={
      Object {
        "fontSize": 15,
        "marginTop": 10,
        "textAlign": "center",
      }
    }
  >
    SecondComponent
  </Text>
  <View>
    <TextInput
      allowFontScaling={true}
      onChangeText={[Function]}
      rejectResponderTermination={true}
      style={
        Object {
          "borderColor": "gray",
          "borderWidth": 1,
          "height": 40,
        }
      }
      testID="input"
      underlineColorAndroid="transparent"
      value=""
    />
    <View
      accessibilityRole="button"
      accessibilityState={Object {}}
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <View
        style={
          Array [
            Object {},
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#007AFF",
                "fontSize": 18,
                "margin": 8,
                "textAlign": "center",
              },
              Object {
                "color": "#841584",
              },
            ]
          }
        >
          Change label visibility
        </Text>
      </View>
    </View>
  </View>
  <RCTScrollView
    data={
      Array [
        Object {
          "data": Array [
            "Pizza",
            "Burger",
            "Risotto",
          ],
          "title": "Main dishes",
        },
        Object {
          "data": Array [
            "French Fries",
            "Onion Rings",
            "Fried Shrimps",
          ],
          "title": "Sides",
        },
        Object {
          "data": Array [
            "Water",
            "Coke",
            "Beer",
          ],
          "title": "Drinks",
        },
        Object {
          "data": Array [
            "Cheese Cake",
            "Ice Cream",
          ],
          "title": "Desserts",
        },
      ]
    }
    disableVirtualization={false}
    getItem={[Function]}
    getItemCount={[Function]}
    horizontal={false}
    initialNumToRender={10}
    keyExtractor={[Function]}
    maxToRenderPerBatch={10}
    onContentSizeChange={[Function]}
    onEndReachedThreshold={2}
    onLayout={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    renderItem={[Function]}
    scrollEventThrottle={50}
    stickyHeaderIndices={
      Array [
        0,
        5,
      ]
    }
    updateCellsBatchingPeriod={50}
    windowSize={21}
  >
    <View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <Text
          style={
            Object {
              "fontSize": 32,
            }
          }
        >
          Main dishes
        </Text>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          style={
            Object {
              "backgroundColor": "#F9C2FF",
              "marginVertical": 8,
              "padding": 20,
            }
          }
        >
          <Text
            style={
              Object {
                "fontSize": 24,
              }
            }
          >
            Pizza
          </Text>
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          style={
            Object {
              "backgroundColor": "#F9C2FF",
              "marginVertical": 8,
              "padding": 20,
            }
          }
        >
          <Text
            style={
              Object {
                "fontSize": 24,
              }
            }
          >
            Burger
          </Text>
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          style={
            Object {
              "backgroundColor": "#F9C2FF",
              "marginVertical": 8,
              "padding": 20,
            }
          }
        >
          <Text
            style={
              Object {
                "fontSize": 24,
              }
            }
          >
            Risotto
          </Text>
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      />
      <View
        onLayout={[Function]}
        style={null}
      >
        <Text
          style={
            Object {
              "fontSize": 32,
            }
          }
        >
          Sides
        </Text>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          style={
            Object {
              "backgroundColor": "#F9C2FF",
              "marginVertical": 8,
              "padding": 20,
            }
          }
        >
          <Text
            style={
              Object {
                "fontSize": 24,
              }
            }
          >
            French Fries
          </Text>
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          style={
            Object {
              "backgroundColor": "#F9C2FF",
              "marginVertical": 8,
              "padding": 20,
            }
          }
        >
          <Text
            style={
              Object {
                "fontSize": 24,
              }
            }
          >
            Onion Rings
          </Text>
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          style={
            Object {
              "backgroundColor": "#F9C2FF",
              "marginVertical": 8,
              "padding": 20,
            }
          }
        >
          <Text
            style={
              Object {
                "fontSize": 24,
              }
            }
          >
            Fried Shrimps
          </Text>
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      />
      <View
        style={
          Object {
            "height": 0,
          }
        }
      />
    </View>
  </RCTScrollView>
</RCTSafeAreaView>
`;
